src/transformers/trainer.py

```diff
@@ -1972,6 +1981,9 @@ def training_step(self, model: nn.Module, inputs: Dict[str, Union[torch.Tensor,
         else:
             loss.backward()
 
+        if is_torch_ltc_available():
+            ltc.mark_step()
```

Comment:
Why did the `mark_step()` location change to here? The previous locations are after `training_step` and `zero_grad`.

Reply:
call `ltc.mark_step` after the loss accumulation caused performance degradation, I'm still looking for the reason, so I moved the `mark_step` in this pr.